nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ValidateJson.java

```diff
@@ -105,97 +140,54 @@ public class ValidateJson extends AbstractProcessor {
 
-    private List<AllowableValue> allowableValues;
-
-    @Override
-    protected void init(final ProcessorInitializationContext context) {
-        final List<PropertyDescriptor> descriptors = new ArrayList<PropertyDescriptor>();
-        descriptors.add(SCHEMA_TEXT);
-        descriptors.add(SCHEMA_VERSION);
-        this.descriptors = Collections.unmodifiableList(descriptors);
-
-        final Set<Relationship> relationships = new HashSet<Relationship>();
-        relationships.add(REL_VALID);
-        relationships.add(REL_INVALID);
-        relationships.add(REL_FAILURE);
-        this.relationships = Collections.unmodifiableSet(relationships);
-
-        final List<AllowableValue> allowableValues = new ArrayList<AllowableValue>();
-        allowableValues.add(SCHEMA_VERSION_4);
-        allowableValues.add(SCHEMA_VERSION_6);
-        allowableValues.add(SCHEMA_VERSION_7);
-        allowableValues.add(SCHEMA_VERSION_V201909);
-        this.allowableValues = Collections.unmodifiableList(allowableValues);
-
-    }
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+    private JsonSchema schema;
 
     @Override
     public Set<Relationship> getRelationships() {
-        return this.relationships;
+        return RELATIONSHIPS;
     }
 
     @Override
     public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
-        return descriptors;
+        return PROPERTIES;
     }
 
-    private ObjectMapper mapper = new ObjectMapper();
-    private VersionFlag schemaVersion;
-
     @OnScheduled
-    public void onScheduled(final ProcessContext context) {
-        // Set JSON schema version to use from processor property
-        this.schemaVersion = VersionFlag.V201909;
-        if (context.getProperty(SCHEMA_VERSION).getValue() == SCHEMA_VERSION_4.getValue()) {
-            this.schemaVersion = VersionFlag.V4;
-        }
-        if (context.getProperty(SCHEMA_VERSION).getValue() == SCHEMA_VERSION_6.getValue()) {
-            this.schemaVersion = VersionFlag.V6;
-        }
-        if (context.getProperty(SCHEMA_VERSION).getValue() == SCHEMA_VERSION_7.getValue()) {
-            this.schemaVersion = VersionFlag.V7;
-        }
-        if (context.getProperty(SCHEMA_VERSION).getValue() == SCHEMA_VERSION_V201909.getValue()) {
-            this.schemaVersion = VersionFlag.V201909;
+    public void onScheduled(final ProcessContext context) throws IOException {
+        try (final InputStream inputStream = context.getProperty(SCHEMA_CONTENT).asResource().read()) {
+            String schemaText = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
+            JsonSchemaVersion jsonSchemaVersion =
+                    JsonSchemaVersion.valueOf(context.getProperty(SCHEMA_VERSION).getValue());
+            JsonSchemaFactory factory = JsonSchemaFactory.getInstance(jsonSchemaVersion.getVersionFlag());
+            schema = factory.getSchema(schemaText);
```

Comment:
@exceptionfactory I just realized that I do not need to do anything with the InputStream as JsonSchemaFactory has a getSchema method which takes an InputStream.

Reply:
Thanks for noting that @dan-s1, passing the `InputStream` directly to `JsonSchemaFactory` would be a helpful improvement.

Reply:
When I commit that change should I squash my commits to one or leave it as two separate commits?

Reply:
A separate commit is best. Squashing the initial commit is recommended, otherwise squashing commits should only be necessary when rebasing.

Reply:
@exceptionfactory Done